packages/flutter_tools/lib/src/commands/multi_drive.dart

```diff
@@ -0,0 +1,375 @@
+        try {
+          await allAppsStopper(this);
+        } catch(error, stackTrace) {
+          // TODO(yjbanov): remove this guard when this bug is fixed: https://github.com/dart-lang/sdk/issues/25862
+          printTrace('Could not stop application: $error\n$stackTrace');
+        }
+      } else {
+        printStatus('Leaving the application running.');
+      }
+    }
+  }
+
+  Future<dynamic> _loadSpecs(String specsPath) async {
+    // Read specs file into json format
+    dynamic spec = JSON.decode(await new io.File(specsPath).readAsString());
+    // Get the parent directory of the specs file
+    String rootPath = new io.File(specsPath).parent.absolute.path;
+    // Normalize the 'test-path' in the specs file
+    spec['test-path'] = _normalizePath(rootPath, spec['test-path']);
+    // Normalize the 'app-path' in the specs file
+    spec['devices'].forEach((String deviceID, Map<String, String> value) {
+      value['app-path'] = _normalizePath(rootPath, value['app-path']);
+    });
+    return spec;
+  }
+
+  String _normalizePath(String rootPath, String relativePath) {
+    return path.normalize(path.join(rootPath, relativePath));
+  }
+}
+
+/// Finds a device to test on. May launch a simulator, if necessary.
+typedef Future<List<Device>> TargetDevicesFinder();
+TargetDevicesFinder targetDevicesFinder = findTargetDevices;
+void restoreTargetDevicesFinder() {
+  targetDevicesFinder = findTargetDevices;
+}
+
+Future<List<Device>> findTargetDevices() async {
+  // Should not specify a single device id
+  /*
+  if (deviceManager.hasSpecifiedDeviceId) {
+    return deviceManager.getDeviceById(deviceManager.specifiedDeviceId);
+  }
+  */
+
+  List<Device> devices = await deviceManager.getAllConnectedDevices();
+
+  if (os.isMacOS || os.isLinux) {
+    // On MacOS or Linux, we grab the all connected device we can find.
+    if (devices.isEmpty) {
+      printError('No devices found.');
+      return null;
+    } else {
+      print(
+        'Found connected device${ devices.length == 1 ? '' : 's' }:\n'
+        '<${devices.join(",")}>'
+      );
+      return devices;
+    }
+  } else if (os.isWindows) {
+    printError('Windows is not yet supported.');
+    return null;
+  } else {
+    printError('The operating system on this computer is not supported.');
+    return null;
+  }
+}
+
+/// Starts the application on the device given command configuration.
+typedef Future<int> MultiDeviceAppsStarter(MultiDriveCommand command);
+
+MultiDeviceAppsStarter appsStarter = startMultiDeviceApps;
+void restoreMultiDeviceAppsStarter() {
+  appsStarter = startMultiDeviceApps;
+}
+
+Device findDevice(List<Device> devices, String deviceID) {
+  for(Device device in devices) {
+    if(device.id == deviceID) return device;
+  }
+  return null;
+}
+
+Future<int> startMultiDeviceApps(MultiDriveCommand command) async {
+  // Load all device ids
+  Map<String, dynamic> devices = command.specs['devices'];
```

Comment:
I was confused and changed the `devices` variable in the `findDevice` function.  Now fixed.